step_definitions/common_steps.py

```diff
@@ -37,6 +39,31 @@ def get_agent_by_driver(driver_):
             return AGENT_CREDENTIALS.get(agent_)
 
 
+def get_agent_by_attributes(user=None, station_id=None, inbound_camp=None, outbound_camp=None, chat_camp=None, ready_for=None):
+    agent = {}
+    for agent_ in AGENT_CREDENTIALS:
+        tmp_agent = AGENT_CREDENTIALS.get(agent_)
+        try:
+            if user:
+                assert tmp_agent.get('user') == user
+            if station_id:
+                assert tmp_agent.get('station').get('station_id') == station_id
+            if inbound_camp:
+                assert tmp_agent.get('inbound_camp') == inbound_camp
+            if outbound_camp:
+                assert tmp_agent.get('outbound_camp') == outbound_camp
+            if chat_camp:
+                assert tmp_agent.get('chat_camp') == chat_camp
+            if ready_for:
+                assert ready_for in tmp_agent.get('ready_channels')
+            agent = tmp_agent
+        except AssertionError:
+            continue
+        except AttributeError:
```

Comment:
Duplicated except clause

Reply:
Actually here we are treating 2 different exceptions.
Assertion when the condition does not match
Attribute when the station_id is not set (for 'None' station option)

Reply:
Read it wrong